### PR TITLE
F-056 uphill lane-marking duty cycle

### PR DESCRIPTION
**Pull request by randroid88**
## Summary
- shorten lane dash visible duty inside the existing road-distance phase cycle
- add renderer geometry tests for in-strip duty boundaries and near-camera short dashes
- add and close F-056 in the backlog and link it to GDD-16-ROAD-MARKINGS

## GDD
- docs/gdd/09-track-design.md
- docs/gdd/16-rendering-and-visual-design.md
- docs/gdd/21-technical-design-for-web-implementation.md

## Progress log
- docs/PROGRESS_LOG.md: 2026-04-27 F-056 uphill lane-marking duty cycle

## Test plan
- npx vitest run src/render/__tests__/pseudoRoadCanvas.test.ts
- npx vitest run src/render/__tests__/pseudoRoadCanvas.test.ts src/road/__tests__/segmentProjector.test.ts
- npm run test:e2e -- e2e/race-demo.spec.ts
- npm run verify
- npm run content-lint
- perl dash scan over touched files
- git diff --check

### CODE REVIEW

File: src/render/pseudoRoadCanvas.ts [hunk 1]
```diff
@@ -777,3 +795,59 @@ function drawPhasedTrapezoids(
...
+  colorAtWorld: (worldZ: number) => string | null,
+  widthOptions: { minNearHalfW?: number; minFarHalfW?: number } = {},
+): void {
+  const span = worldFar - worldNear;
+  if (span <= 0) return;
+  if (cycleMeters <= 0 || visibleMeters <= 0 || visibleMeters >= cycleMeters) {
+    drawPhasedTrapezoids(
+      ctx,
+      near,
+      far,
```
[the remark this answers is not in the document]
Comment (randroid88) [reply src/render/pseudoRoadCanvas.ts]: Fixed in e15b9d6: non-positive cycle or visible lengths now return before phase math, and the fallback only runs for positive full-duty cycles.
